main.tf

```diff
@@ -10,6 +10,7 @@ locals {
   create_nat_gateway            = (var.nat != "NO_NAT" && var.nat != "NAT_INSTANCES") ? true : false
   create_single_nat_gateway     = (var.nat == "NAT_GATEWAY_SINGLE") ? true : false
   create_one_nat_gateway_per_az = (var.nat == "NAT_GATEWAY_PER_AZ") ? true : false
+  create_alter_nat              = (var.nat == "ALTER_NAT") ? true : false
```

Comment:
Bug: this also creates a NAT Gateway. See line 10. Maybe you can rewrite line 10 in a way that it's easier to understand and that future additions won't break with line 10.